specs/master/cluster-init/files/host_provider/src/cluster.py

```diff
@@ -3,14 +3,35 @@
 import logging
 from urllib.parse import urlencode
 from builtins import str
-
+from copy import deepcopy
 
 try:
     import cyclecli
 except ImportError:
     import cyclecliwrapper as cyclecli
     
 
+def limit_request_by_available_count(status, request, logger):
+        request_copy = deepcopy(request)
+        request_set = request_copy['sets'][0]
+        machine_type = request_set["definition"]["machineType"]
+        nodearray_name = request_set['nodearray']
+        filtered = [x for x in status["nodearrays"] if x["name"] == nodearray_name]
+        assert len(filtered) == 1
```

Comment:
Before release, we should remove the asserts in this method and replace them with an error log and a reasonable alternative (like making no changes or a RuntimeError with reasonable message if they're an expected error condition)